### PR TITLE
Automapper Fix when using AutoMapper ProjectTo in LINQ query

### PR DESCRIPTION
**Pull request by michielpeeters**
This piece of code fixes the problem when AutoMapper extension ProjectTo is being used.

### CODE REVIEW

**Comment by huysentruitw** [comment]
What is the exact error you're trying to fix here?

File: src/EntityFrameworkMock.Shared/DbAsyncQueryProvider.cs [hunk 1]
```diff
@@ -16,7 +17,21 @@ public DbAsyncQueryProvider(IQueryProvider inner)
             _inner = inner;
         }
 
-        public IQueryable CreateQuery(Expression expression) => new DbAsyncEnumerable<TEntity>(expression);
+        public IQueryable CreateQuery(Expression expression)
+        {
+            switch (expression)
```
Comment (huysentruitw) [inline src/EntityFrameworkMock.Shared/DbAsyncQueryProvider.cs]: A switch should not be needed for only testing one case.

File: src/EntityFrameworkMock.Shared/DbAsyncQueryProvider.cs [hunk 2]
```diff
@@ -16,7 +17,21 @@ public DbAsyncQueryProvider(IQueryProvider inner)
             _inner = inner;
         }
 
-        public IQueryable CreateQuery(Expression expression) => new DbAsyncEnumerable<TEntity>(expression);
+        public IQueryable CreateQuery(Expression expression)
```
Comment (huysentruitw) [inline src/EntityFrameworkMock.Shared/DbAsyncQueryProvider.cs:20]: Please add one or more unit-tests to cover this change.

**Review by huysentruitw — CHANGES_REQUESTED** [review]

**Comment by michielpeeters** [comment]
You are absoluty right :-) sorry for that! The error which is thrown is:

Unable to cast object of type 'EntityFrameworkMock.DbAsyncEnumerable 1[EntityFrameworkMock.Tests.Models.User] to type System.Linq.IQueryable 1[EntityFrameworkMock.Tests.DbContextMockTests+UserViewModel]'.

And is fixed with this PR. I've added a testcase ofcourse :-).

**Comment by huysentruitw** [comment]
I'll merge it to the develop branch and do some small changes before merging it to the master branch.

Thanks for this change!

**Comment by michielpeeters** [comment]
You're welcome!

**Pull request merged**